### ISSUE TITLE
register usage incorrectly produces a warning and suggests [[vk::binding(...)]]

### ISSUE THREAD

**Issue by stanoddly** [shader-slang/slang#5938]
If I try to compile this shader:
```
Sampler2D CombinedSampler : register(t0): register(s0);

[shader("fragment")]
float4 main(float2 TexCoord : TEXCOORD0) : SV_Target0
{
    return CombinedSampler.Sample(TexCoord);
}
```

via command line:
```
/opt/slang/bin/slangc shader.slang -target "spirv" -o "shader.spv"
```

it seems to generate `shader.spv` as expected, but also a warning is produced:
```
 warning 39013: shader parameter 'CombinedSampler' has a 'register' specified for D3D, but no '[[vk::binding(...)]]` specified for Vulkan
```

However, such usage is explicitly documented in "SPIR-V specific functionalities" ([link](shader-slang.com/slang/user-guide/spirv-target-specific.html#combined-texture-sampler)):

>  ## Combined texture sampler
> 
> Slang supports Combined texture sampler such as `Sampler2D`. Slang emits SPIR-V code with `OpTypeSampledImage` instruction.
> 
> You can specify two different register numbers for each: one for the texture register and another for the sampler register.
> 
> ```
> Sampler2D explicitBindingSampler : register(t4): register(s3);
> ```
> 

So, there should be no warning for a `register` usage, shouldn't be? I know the warning could be disabled via `-warnings-disable`, however I would expect there is nothing to warn about or alternatively register usage shouldn't be documented.

I would personally say that any usage of register shouldn't produce any warning. I think the behavior should be aligned with how `dxc` - `dxc` produces no warnings when compiling to SPIR-V ([source](https://github.com/microsoft/DirectXShaderCompiler/blob/main/docs/SPIR-V.rst#implicit-binding-number-assignment) + verified manually).

**Comment by csyonghe** [comment]
We should remove this warning now that we now support the same -fvk-shift options as in DXC.

**Comment by cheneym2** [comment]
@csyonghe What is the syntax in https://github.com/orgs/shader-slang/projects/10/views/shader-slang.com/slang/user-guide/spirv-target-specific.html#combined-texture-sampler supposed to do for a SPIRV target?

**Comment by csyonghe** [comment]
A Sampler2D maps to a single combined texture sampler descriptor in Vulkan, so it will have only one binding index. This is basically what sampler2D in glsl maps to.

**Comment by cheneym2** [comment]
@csyonghe My understanding is that we should:

1) Keep the warning as-is for the case cited in this issue
2) Improve the documentation around the feature by
* adding a [[vkbinding]] in the example
* noting that targets which don't support combined texture samplers can be bound with separate register assignments

**Comment by csyonghe** [comment]
Agreed. We can improve the error message to ask the user to either specify [vk::binding] or specify `-fvk-xxx-shift`.

**Comment by cheneym2** [comment]
https://github.com/shader-slang/slang/pull/6207 is currently in Draft state.

I intend to make it so the warning in this bug is _not_ silenced by adding irrelevant -vk-xxx-shift options, but the new test that enforces that is not passing yet.

**Comment by cheneym2** [comment]
https://github.com/shader-slang/slang/pull/6207 is ready for review
